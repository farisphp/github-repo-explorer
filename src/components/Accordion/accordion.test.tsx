import { render, screen } from '@testing-library/react'
import { Accordion, AccordionContent, AccordionItem, AccordionTrigger } from '.'

describe('<Accordion />', () => {
  it('should render accordion correctly', () => {
    const { container } = render(
      <Accordion type="single" collapsible className="w-full">
        <AccordionItem value="item-1">
          <AccordionTrigger>Is it accessible?</AccordionTrigger>
          <AccordionContent>
            Yes. It adheres to the WAI-ARIA design pattern.
          </AccordionContent>
        </AccordionItem>
      </Accordion>
    )

    const heading = screen.getByRole('heading', {
      name: 'Is it accessible?',
      level: 3
    })
    expect(heading).toBeInTheDocument()

    const accordionTrigger = screen.getByRole('button', {
      name: 'Is it accessible?'
    })
    expect(accordionTrigger).toBeInTheDocument()
    expect(accordionTrigger).toHaveAttribute('aria-expanded', 'false')

    expect(container.firstChild).toMatchSnapshot()
  })
})
