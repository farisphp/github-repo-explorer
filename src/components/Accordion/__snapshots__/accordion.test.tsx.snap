// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Accordion /> > should render accordion 1`] = `
<div
  class="w-full"
  data-orientation="vertical"
>
  <div
    class="border-b"
    data-orientation="vertical"
    data-state="closed"
  >
    <h3
      class="flex"
      data-orientation="vertical"
      data-state="closed"
    >
      <button
        aria-controls="radix-:r1:"
        aria-expanded="false"
        class="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180"
        data-orientation="vertical"
        data-radix-collection-item=""
        data-state="closed"
        id="radix-:r0:"
        type="button"
      >
        Is it accessible?
        <svg
          class="h-4 w-4 transition-transform duration-200"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
          />
        </svg>
      </button>
    </h3>
    <div
      aria-labelledby="radix-:r0:"
      class="overflow-hidden text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down"
      data-orientation="vertical"
      data-state="closed"
      hidden=""
      id="radix-:r1:"
      role="region"
      style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
    />
  </div>
</div>
`;

exports[`<Accordion /> > should render accordion correctly 1`] = `
<div
  class="w-full"
  data-orientation="vertical"
>
  <div
    class="border-b"
    data-orientation="vertical"
    data-state="closed"
  >
    <h3
      class="flex"
      data-orientation="vertical"
      data-state="closed"
    >
      <button
        aria-controls="radix-:r1:"
        aria-expanded="false"
        class="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180"
        data-orientation="vertical"
        data-radix-collection-item=""
        data-state="closed"
        id="radix-:r0:"
        type="button"
      >
        Is it accessible?
        <svg
          class="h-4 w-4 transition-transform duration-200"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
          />
        </svg>
      </button>
    </h3>
    <div
      aria-labelledby="radix-:r0:"
      class="overflow-hidden text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down"
      data-orientation="vertical"
      data-state="closed"
      hidden=""
      id="radix-:r1:"
      role="region"
      style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
    />
  </div>
</div>
`;
